Add s2n_config_set_extension_data method to add/change TLS extension data

baldwinmatt opened the pull request:
Add s2n_config_set_extension_data method to add/change TLS extension data after creating the initial configuration.

This is useful for extensions such as OCSP where the data needs to be refreshed over time, if the OCSP status expires or is updated, for example.

Closes #461

raycoll commented on api/s2n.h:
```diff
@@ -73,15 +76,14 @@ extern int s2n_config_add_cert_chain_and_key_with_extensions(struct s2n_config *
         const char *private_key_pem, s2n_tls_extension *extensions, uint16_t num_extensions);
```
Do we still need `s2n_config_add_cert_chain_and_key_with_extensions` if we have `s2n_config_set_extensions`? I don't think it's been in the API long enough to have backward-compatibility concerns.

baldwinmatt replied:
Sounds reasonable.  I'll make that chnage.

baldwinmatt commented:
Cleaned this up a bit:

- Removed the `_with_extensions` API method, as it was only recently introduced and this change makes it unnecessary
- Removed `s2n_tls_extension` from the API as it was only really needed if we were using the `_with_extensions` API.  This closes #461

raycoll commented:
Nice simplification 😄
